packages/td-tools/src/util/asset-interface-description.ts

```diff
@@ -359,12 +359,25 @@ export class AssetInterfaceDescriptionUtil {
             for (const entry of smInformation.properties.entries()) {
                 const key = entry[0];
                 const value: AASInteraction[] = entry[1];
-                logInfo("Property" + key + " = " + value);
+                logInfo("vProperty" + key + " = " + value);
 
                 thing.properties[key] = {};
                 thing.properties[key].forms = [];
 
                 for (const vi of value) {
+                    // The first if condition is expected to be temporary. will be adjusted or removed when a decision on how the datapoint's datatype would be modelled is made for AID
+
+                    if (vi.interaction.constraints instanceof Array && vi.interaction.constraints) {
```

Comment:
I believe the order of the checks should change, first null/undefined check and then whether it is array.